pdf/pdfcanvas.cpp

```diff
@@ -164,12 +164,14 @@ void PDFCanvas::setDocument(PDFDocument* doc)
     }
 }
 
-qreal PDFCanvas::pagePosition(int index) const
+qreal PDFCanvas::pagePosition(int index, qreal offset) const
 {
     if( d->pages.count() == 0 )
         return 0.f;
 
-    return d->pages.value( index ).rect.y();
+    const QRectF& rect = d->pages.value( index ).rect;
+    qreal minOffset = 0., maxOffset = 1.;
+    return rect.y() + rect.height() * qBound(minOffset, offset, maxOffset);
```

Comment:
The later sugestion was replacing qreal pagePosition(int) with QRectF pageRectangle(int). That gets converted into qml rect type and code in PDFView.qml can decide what position between rect.y and (rect.y + rect.height) to set.